flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertex.java

```diff
@@ -307,13 +323,21 @@ public JobVertexID getJobVertexId() {
 
     @Override
     public ExecutionVertex[] getTaskVertices() {
+        if (taskVertices == null) {
```

Comment:
Why not use `checkState(isInitialized()); ` to check whether ExecutionJobVertex is initialized?

Reply:
It is not for unit testing. The REST API calls this method when the job vertex is not initialized. So we just print log instead of throw an exception here. BTW, the log here should be changed to DEBUG.